exercises/ex1.cpp

```diff
@@ -1,48 +1,66 @@
 /* Improve this program using a switch-case. */
 
 #include <iostream>
+#include <limits>
 using namespace std;
 
+template <class Type>
+Type insert_and_control(Type& data) //cin control
+{
+	do
+	{
+		cin >> data;
+		if (cin.fail())
+		{
+			cerr << "Insert a number!" << endl;
+			cin.clear();
+			cin.ignore(numeric_limits<streamsize>::max(), '\n');
+		}
+		else
+			break;
+	} while (true);
+	return data;
+}
+
 int main()
 {
-  int week;
+	int week;
 
-  /* Input week number from user */
-  cout << "Enter week number(1-7): " << endl;
-  cin >> week;
+	/* Input week number from user */
+	cout << "Enter week number(1-7): " << endl;
+	do
+	{
 
-  if (week == 1)
-  {
-    cout << "Monday" << endl;
-  }
-  else if (week == 2)
-  {
-    cout << "Tuesday" << endl;
-  }
-  else if (week == 3)
-  {
-    cout << "Wednesday" << endl;
-  }
-  else if (week == 4)
-  {
-    cout << "Thursday" << endl;
-  }
-  else if (week == 5)
-  {
-    cout << "Friday" << endl;
-  }
-  else if (week == 6)
-  {
-    cout << "Saturday" << endl;
-  }
-  else if (week == 7)
-  {
-    cout << "Sunday" << endl;
-  }
-  else
-  {
-    cout << "Invalid input! Please enter week number between 1-7." << endl;
-  }
+		week = insert_and_control(week); 
```

Comment:
converti la funzione in una funzione che ritorna void, rimuovi l'assegnazione qui, adesso non serve più
```c++
insert_and_control(week)
```

Reply:
fatto!